modules/system-view-api/README.md

```diff
@@ -0,0 +1,127 @@
+            .build();
+
+    SystemView<ClusterNode> localView = SystemViews.<ClusterNode>nodeViewBuilder()
+            .name("TOPOLOGY_SNAPSHOT_LOCAL_VIEW")
+            .nodeNameColumnAlias("VIEW_DATA_SOURCE")
+            .addColumn("NAME", NativeTypes.STRING, ClusterNode::name)
+            .addColumn("HOST", NativeTypes.STRING, node -> node.address().host())
+            .addColumn("PORT", NativeTypes.INT32, node -> node.address().port())
+            .dataProvider(SubscriptionUtils.fromIterable(() -> ignite.iterator()))
+            .build();
+            
+    return List.of(clusterWideView, localView);
+}
+```
+
+### Step 3. Register your provider in views manager
+
+Component `SystemViewManager` is responsible for views registration in the system.
+Currently there is no automatic registration of system components views, so we need to do it manually.
+```java
+MyComponent component = new MyComponent();
+systemViewManager.register(component);
+```
+Please note that this must be done before `systemViewManager` starts.
+
+### Check the result
+
+After these steps, we can start the cluster and start working with the system views we just created.
+
+Note that the list of registered views can be obtained using the `SYSTEM_VIEWS` view.
+
+```sql
+select * from system.system_views order by id
+```
+| ID | SCHEMA | NAME                         | TYPE    |
+|----|--------|------------------------------|---------|
+| 3  | SYSTEM | SYSTEM_VIEWS                 | CLUSTER |
+| 4  | SYSTEM | SYSTEM_VIEW_COLUMNS          | CLUSTER |
+| 5  | SYSTEM | TOPOLOGY_SNAPSHOT            | CLUSTER |
+| 6  | SYSTEM | TOPOLOGY_SNAPSHOT_LOCAL_VIEW | NODE    |
+
+#### Check cluster-wide system view
+
+```sql
+select * from system.topology_snapshot
+```
+| host     | address       | port |
+|----------|---------------|------|
+| isvt_n_0 | 192.168.0.103 | 3344 |
+| isvt_n_1 | 192.168.0.103 | 3345 |
+| isvt_n_2 | 192.168.0.103 | 3346 |
+
+For a cluster-wide system view, the data is retrieved on one of the nodes on which the view was registered.
+
+#### Check local node system view
+
+```sql
+select * from system.topology_snapshot_local_view
+```
+| view_data_source | host      | address       | port |
+|------------------|-----------|---------------|------|
+| isvt_n_0         | isvt_n_0  | 192.168.0.103 | 3344 |
+| isvt_n_0         | isvt_n_1  | 192.168.0.103 | 3345 |
+| isvt_n_0         | isvt_n_2  | 192.168.0.103 | 3346 |
+| isvt_n_2         | isvt_n_0  | 192.168.0.103 | 3344 |
+| isvt_n_2         | isvt_n_1  | 192.168.0.103 | 3345 |
+| isvt_n_2         | isvt_n_2  | 192.168.0.103 | 3346 |
+| isvt_n_1         | isvt_n_0  | 192.168.0.103 | 3344 |
+| isvt_n_1         | isvt_n_1  | 192.168.0.103 | 3345 |
+| isvt_n_1         | isvt_n_2  | 192.168.0.103 | 3346 |
+
+Since the local cluster includes 3 nodes, the data is duplicated three times.
+Because for a local view, data is retrieved on **each node** where the view is registered.
```

Comment:
Should we add additional sections with some rocomndations and best approaches with working with the system views?
Just my thoughts:
1) Use concrete class in generic instead of `?`, like a SystemViews.<ParentIdAwareDescriptor<CatalogTableColumnDescriptor>>clusterViewBuilder()
2) Use upper case for name of view and their columns + may be add information about the no ability to use case sensitive mode
3) where we usaly register new views
......

Reply:
> Use upper case for name of view

Added clarification about case insensitivity names in the "Basic concepts" section.

> Use concrete class in generic instead of ?

To be honest, I was unable to specify the provider and data provider without specifying a specific class.
The only thing you can do is not specify generic (that is, keep Object), but in this case, firstly, a warning is highlighted and secondly, you will not be able to extract data for the column.

> where we usaly register new views

I'm not sure I understand this :thinking:

Reply:
ok, thanks for the explanation, you understand me correct.

Reply:
Added note about that
```
* All system views must be registered during node startup. The `IgniteImpl` constructor is currently used as a starting point to do this.
```
:thinking: :thinking:

Reply:
it worth to mention though that it safer to explicitly specify type of the column:

```
SystemViews.clusterViewBuilder()
                .<Integer>addColumn("VIEW_ID", NativeTypes.INT32, entry -> entry.id)
```

in the example above, there is no way the type of `entry.id` will be changed silently

Reply:
The example has been updated with column type parameterization and a note about this has been added.